backend/api/models/computeplan.py

```diff
@@ -98,17 +96,15 @@ def update_status(self) -> None:
         """Compute cp status from tasks counts."""
         stats = self.get_task_stats()
         if stats["task_count"] == 0:
-            compute_plan_status = self.Status.PLAN_STATUS_EMPTY
+            compute_plan_status = self.Status.PLAN_STATUS_CREATED
+        elif stats["waiting_builder_slot_count"] == stats["task_count"]:
+            compute_plan_status = self.Status.PLAN_STATUS_CREATED
```

Comment:
You could group the two in the same `if`.

Also, the `self.Status.PLAN_STATUS_TODO` was completely removed. Is that on purpose ? If yes, why ?

Reply:
Yeah I will merge it if you think it is cleaner!

So for the status changes:
- `EMPTY` and `WAITING` have been merged into `CREATED` which states only the existence of CP (EMPTY being just CREATED with 0 tasks)
- `WAITING` lost it's meaning by itself (waiting what? the build? the execution?)
- `TODO` was a reflection of a compute task being ready to be ran (built + execution). FOr compute task statuses, this status became `WAITING_FOR_EXECUTOR_SLOT`. and is not the first action done on the task. With product, we didn't think that the compute plan status needed the granularity of compute tasks (i.e. needed 3 types of waiting) and a compute plan could also have tasks in different waiting states at the same time (therefore what todo would mean? One task has to be built? executed? etc)

Reply:
ok, so you mean that the case `stats["doing_count"] == 0 and stats["done_count"] == 0:` is in fact that a task is `WAITING_FOR_EXECUTOR_SLOT`, so the CP should still be in `DOING` ?

Reply:
Yes, because `BUILDING` is part of doing. That's why we renamed `DOING` in `EXECUTING`, because `DOING` encompasses `BUILDING` and `EXECUTING`. I don't think we could reach a state where we have neither 

You are raising an interesting point, I think I should revert renaming `doing_count`  to `executing_count` and adding `BUILDING` as part of it

Reply:
After second thoughts, it seems like we should keep `executing_count` as these metrics refers to task statuses (explicitely providing a count for each status) rather than a compute task one